### ISSUE TITLE
Setting lines.markeredgecolor in rcParams affects the ticks' mark color too

### ISSUE THREAD

**Issue by marccheneau** [matplotlib/matplotlib#14546]
### Bug report

**Bug summary**

Setting "lines.markeredgecolor" in rcParams affects the ticks' mark color too. This is unexpected I guess.

Using "axes.prop_cycle" and the cycler label "markeredgecolor" produces the expected result, that is, it only changes the marker edge color and not the ticks' mark color.

**Code for reproduction**

```python
import numpy as np
import matplotlib
import matplotlib.pyplot as plt

matplotlib.rcParams["lines.markeredgecolor"] = "red"

x = np.linspace(0, 10)
plt.plot(x, x**2, marker="o")
plt.show()
```

**Actual outcome**

The ticks' mark color has turned red, together with the marker edge color.

**Expected outcome**

The ticks' mark color should ramin to its default value.

**Matplotlib version**
  * Operating system: Archlinux
  * Matplotlib version: 3.1.0 (installed using the Archlinux's package manager)
  * Matplotlib backend: Qt5Agg
  * Python version: 3.7.3

**Comment by WeatherGod** [comment]
`lines.markeredgecolor` in the rcParams is for *any* line created in the
figure, so that includes the axis spines. The tickmarkers are implemented
as markers of the axis spine. The property cycler applies only to the
things produced by the plotting calls.

**Comment by WeatherGod** [comment]
At the moment, I am inclined to close this as "not a bug", but I could be convinced otherwise.

**Comment by timhoffm** [comment]
I disagree. Though technically, the ticks may be implemented as markers, conceptually, they are quite different from markers for data points (and should probably almost always have different properties). I could otherwise argue that ticks are lines and should follow all the `lines.*`properties, which they don't.

Ticks are an independent entity and should be treated as such. They also already follow `xtick.color`. Having both is rather confusing (even though `xtick.color` simultaneously colors the tick and the tick label).

**Comment by marccheneau** [comment]
I tried to globally (meaning in the rcParams) and independently set the color of the "data markers" and of the ticks' marks using `lines.markeredgecolor` together with `xtick.color` but this doesn't work: the color of the ticks' marks is bound to that of the data markers.

It does remain possible to set the data marker's color alone when calling the plot function, but I keep thinking that the possibility for a global setting would be desirable too. It's not a bug, right, but maybe a missing feature.

**Comment by QuLogic** [comment]
This was fixed by #16847.
